ddtrace/tracer.py

```diff
@@ -662,8 +670,9 @@ def _initialize_span_processors(self):
         # type: () -> None
         trace_processors = []  # type: List[TraceProcessor]
         trace_processors += [TraceTagsProcessor()]
-        trace_processors += [TraceSamplingProcessor()]
-        trace_processors += [TraceTopLevelSpanProcessor()]
+        trace_processors += [TraceSamplingProcessor(self._compute_stats)]
```

Comment:
We can move the sampling processor first to save any processing if the trace will be dropped.

Reply:
Yes, I think so.